nextcord/client.py

```diff
@@ -329,6 +328,8 @@ def __init__(
         self._ready: asyncio.Event = asyncio.Event()
         self._connection._get_websocket = self._get_websocket
         self._connection._get_client = lambda: self
+        self._token: str | None = None
```

Comment:
As we still target a version lower then 3.10 this will need to change (I think, futures may fix it)
```suggestion
        self._token: Optional[str] = None
```
Note: Check optional is also imported